common/lib/user.js

```diff
@@ -1,25 +1,63 @@
-function User (token) {
-  const permissions = []
-  const authorities = token.authorities || []
-
-  if (authorities.includes('ROLE_PECS_POLICE')) {
-    permissions.push(
-      ...[
-        'moves:view:by_location',
-        'moves:download:by_location',
-        'move:view',
-        'move:create',
-        'move:cancel',
-      ]
-    )
-  }
-
-  if (authorities.includes('ROLE_PECS_SUPPLIER')) {
-    permissions.push(...['moves:view:all', 'moves:download:all'])
-  }
-
-  this.userName = token.user_name
-  this.permissions = permissions
+const Sentry = require('@sentry/node')
+
+const referenceDataService = require('../../common/services/reference-data')
+// TODO: Remove once we can determine locations from auth
+const { CURRENT_LOCATION_UUID } = require('../../config')
+
+function User () {}
+
+User.prototype = {
+  init (token) {
+    return new Promise((resolve, reject) => {
+      this.userName = token.user_name
+      this.permissions = this.getPermissions(token.authorities)
+
+      this.getLocations(token.locations)
+        .then(locations => (this.locations = locations.filter(Boolean)))
+        .then(resolve)
+        .catch(reject)
+    })
+  },
+  getPermissions (authorities = []) {
+    const permissions = []
+
+    if (authorities.includes('ROLE_PECS_POLICE')) {
+      permissions.push(
+        ...[
+          'moves:view:by_location',
+          'moves:download:by_location',
+          'move:view',
+          'move:create',
+          'move:cancel',
+        ]
+      )
+    }
+
+    if (authorities.includes('ROLE_PECS_SUPPLIER')) {
+      permissions.push(...['moves:view:all', 'moves:download:all'])
+    }
+
+    return permissions
+  },
+  getLocations (locations = []) {
+    // TODO: Remove once we can determine locations from auth
+    locations.push(CURRENT_LOCATION_UUID)
+
+    const locationPromises = locations.map(locationId => {
+      return referenceDataService
+        .getLocationById(locationId)
```

Comment:
The value we receive from HMPPS SSO will be the NOMIS agency ID, not the location ID as known by the reference data service. We probably want a service method like `getLocationByNomisAgencyId` to be clear what it is.

Also, the values we get in the token will be prepended with `PECS_` so we'll need to strip these off the value beforehand.

Reply:
Yep, I thought that might be the case. Right now this just moves the location to the user object so that once we get it back we can replace it in one place.

It temporarily still uses the ID from config which we can remove once we get an actual list back from SSO.

The reason for moving this here was so that the user object has some concept of allowed locations so that we can check that they can see the location they might be viewing in the moves feature.

Reply:
tl;dr I think that will be a future change

Reply:
I think given we're introducing the `getLocations` method in this PR we should make sure it's not a knowingly incorrect implementation... we can set the environment variable to `PECS_GDI` now and get that working.

Reply:
The issue with that is that it also involves an API change so will hold up the other changes that are part of this PR, like being able to associate locations to a user.

Reply:
We also don't know the exact implementation of how it will come back as part of the auth yet so might be better to leave that change once we have an example implementation of it.

Reply:
This is already implemented on the SSO service.

Reply:
As discussed on slack I think we need to leave the implementation of SSO `agencyId` support for a future change once we also have support in the API to get a location by agencyId.

This is holding up some other changes that we need in soon and the main basis for this is to move the locations to the user object so that other parts of the app can use this.

Reply:
Backend ticket: https://dsdmoj.atlassian.net/browse/P4-511
Frontend ticket: https://dsdmoj.atlassian.net/browse/P4-512